main/src/main/java/org/mobilitydata/gtfsvalidator/cli/Main.java

```diff
@@ -75,29 +78,40 @@ public static void main(String[] argv) {
         gtfsInput = GtfsInput.createFromPath(Paths.get(args.getInput()));
       }
     } catch (IOException | URISyntaxException | InterruptedException e) {
-      // FIXME: Add system errors to noticeContainer here.
+      if (e instanceof IOException) {
+        noticeContainer.addSystemError(new IOError(e.getMessage()));
+      }
+      if (e instanceof URISyntaxException) {
+        noticeContainer.addSystemError(new URISyntaxError(e.getMessage()));
+      } else {
+        noticeContainer.addSystemError(new ThreadInterruptedError(e.getMessage()));
+      }
+      exportReport(args.getOutputBase(), noticeContainer);
```

Comment:
Please write separate `catch` sections for each type.

```
} catch (IOException e) {
  logger.atSevere().withCause(e).log("Cannot load GTFS feed");
  noticeContainer.addSystemError(new IOError(e.getMessage()));
} ... {
} ... {
}

if (gtfsInput == null) {
   exportReport(args.getOutputBase(), noticeContainer);
   return;
}
```

Reply:
Thanks, I was actually wondering what would be the best practice in this case.